### PR TITLE
Update samples

### PR DESCRIPTION
**Pull request by twsouthwick**
This change does the following:

- Create a .slnLaunch file to easily launch the samples (including the multi-app samples)
- Target .NET 8.0 for samples
- Update YARP to 2.1.0
- Clean up registrations to recommended patterns
- Convert samples to use MSBuild.SDK.SystemWeb
- Directory.Build.[props|targets] for samples can now bring in the arcade sdk for more consistent projects

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     }
   },
   "msbuild-sdks": {
+    "MSBuild.SDK.SystemWeb": "4.0.88",
```
Comment (joperezr) [inline global.json:21]: How will this get updated when new versions are available? Do we care at all?

**Review by joperezr — APPROVED** [review]

File: samples/Directory.Build.props [hunk 2]
```diff
@@ -1,7 +1,53 @@
...
+    <!-- MSBuild.SDK.SystemWeb tries to bring in default versions of the compilers, but that is now handled automatically by .NET 9 SDK -->
+    <ExcludeSDKDefaultPackages>true</ExcludeSDKDefaultPackages>
+
+    <!-- This allows the first build to be correct without any warnings on a new clone of the repo -->
+    <GeneratedBindingRedirectsAction>Overwrite</GeneratedBindingRedirectsAction>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <NoWarn>
+      $(NoWarn);
```
Comment (joperezr) [inline samples/Directory.Build.props:17]: NIT: typically you would instead create an .editorconfig or .ruleset where you define which warnings should trigger.

Comment (twsouthwick) [reply samples/Directory.Build.props:17]: I agree, but with this it was quick and dirty to remove warnings and I'll follow up with a proper fix to the code patterns

File: global.json (hunk 1 above)
Comment (twsouthwick) [reply global.json:21]: Not sure - probably manually until dependabot can handle global.json

**Pull request merged**